### PR TITLE
JetBrains: bump kotlin to 1.9.10

### PR DESCRIPTION
**Pull request by Strum355**
Unlocks indexing of the plugin with scip-java (see [here for example](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@b71b22ef9584dc8441f9195f2decd52a53a9e2fb/-/blob/client/jetbrains/src/main/java/com/sourcegraph/cody/agent/CodyAgentCodebase.kt?L10:7&popover=pinned#tab=references)). 

## Test plan

`./gradlew compileKotlin` and (hopefully) covered by CI too

### CODE REVIEW

File: client/jetbrains/src/main/kotlin/com/sourcegraph/cody/ui/SelectOptionPopupUI.kt [hunk 1]
```diff
@@ -212,6 +212,7 @@ class SelectOptionPopupUI(
     return DefaultListCellRenderer()
   }
 
+  @Deprecated("Deprecated in Java")
```
Comment (Strum355) [inline client/jetbrains/src/main/kotlin/com/sourcegraph/cody/ui/SelectOptionPopupUI.kt:215]: https://youtrack.jetbrains.com/issue/KT-47902

File: client/jetbrains/src/main/kotlin/com/sourcegraph/cody/ui/SelectOptionPopupUI.kt [hunk 2]
```diff
@@ -85,7 +85,7 @@ class SelectOptionPopupUI(
     get() = mySearchField
 
   private fun executeCommand() {
-    val selectedValue = myResultsList.selectedValue as String ?: return
+    val selectedValue = (myResultsList.selectedValue ?: return) as String
```
Comment (Strum355) [inline client/jetbrains/src/main/kotlin/com/sourcegraph/cody/ui/SelectOptionPopupUI.kt:88]: `as String ?: return` would never hit the return case, Im assuming `myResultsList.selectedValue` can be nullable here, or do we assume it to always be `String`?

Comment (szymonprz) [reply client/jetbrains/src/main/kotlin/com/sourcegraph/cody/ui/SelectOptionPopupUI.kt:88]: You're right, by @Gedochao is going to remove this whole class anyway as it's a dead code

**Review by szymonprz — APPROVED** [review]
Everything seems to work with `./gradlew runPluginVerifier`

**Review by olafurpg — APPROVED** [review]

**Pull request merged**